File: src/main/java/com/freya02/botcommands/api/components/event/ButtonEvent.java
package com.freya02.botcommands.api.components.event;

import com.freya02.botcommands.api.BContext;
import com.freya02.botcommands.internal.BContextImpl;
import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;

public class ButtonEvent extends ButtonInteractionEvent {
	private final BContext context;

	public ButtonEvent(BContextImpl context, ButtonInteractionEvent event) {
		super(event.getJDA(), event.getResponseNumber(), event.getInteraction());

		this.context = context;
	}

	public BContext getContext() {
		return context;
	}
}

File: src/main/java/com/freya02/botcommands/api/components/event/ButtonEvent.kt
package com.freya02.botcommands.api.components.event

import com.freya02.botcommands.api.BContext
import com.freya02.botcommands.api.commands.ratelimit.CancellableRateLimit
import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent

class ButtonEvent internal constructor(
    val context: BContext,
    event: ButtonInteractionEvent,
    cancellableRateLimit: CancellableRateLimit
) : ButtonInteractionEvent(event.jda, event.responseNumber, event.interaction),
    CancellableRateLimit by cancellableRateLimit
